core/currentLimiter.go

```diff
@@ -0,0 +1,121 @@
+package core
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/evcc-io/evcc/util"
+)
+
+// currentLimiter abstracts the LM currentLimiter implementation
+// TBD: required?
+// type currentLimiter interface {
+// 	Enable(enable bool)                                             // enable/disable limitting
+// 	SetTotalLimit(current float64)                                  // set the total limit of this limiter. 0 = unlimited
+// 	SetMargin(current float64)                                      // sets the limit by providing the margin to current consumption
+// 	EvaluateConsumption() float64                                   // evaluates the actual current consumption of
+// 	RequestCurrent(lp *LoadPoint, currentRequested float64) float64 // request current, return limit <= currentRequested
+// }
+
+type CurrentLimiter struct {
+	title      string // meaningful name
+	log        *util.Logger
+	uiChan     chan<- util.Param
+	enabled    bool         // apply limitting
+	maxCurrent float64      // the max allowed current of this limiter
+	loadpoints []*LoadPoint // all loadpoints under management
+}
+
+// ------- interface limiter
+func (lm *CurrentLimiter) Enable(enable bool) {
+	lm.enabled = enable
+	lm.publish("enabled", lm.enabled)
+}
+
+func (lm *CurrentLimiter) SetTotalLimit(current float64) {
+	if current != lm.maxCurrent {
+		lm.log.INFO.Printf("set limit: %.1f", current)
+		lm.maxCurrent = current
+		lm.publish("maxCurrent", lm.maxCurrent)
+	}
+}
+
+// allow this delta current related to actual consumption of all lp
+func (lm *CurrentLimiter) SetMargin(current float64) {
```

Comment:
Was tut das?

Reply:
Korregiert den Limiter um den `current`. Wird nur im GridLimiter der site benutzt. Kann weg, dann muss aber der Summenstrom aller LP in der site ermittelt werden, damit dann im `Limiter.SetTotalLimit()` benutzt werden kann.

Reply:
Removed, obsolete mit meter